osu.Game.Rulesets.Osu/OsuInputManager.cs

```diff
@@ -43,17 +53,52 @@ protected override bool Handle(UIEvent e)
             return base.Handle(e);
         }
 
+        private OsuAction getActionForTouchSource(TouchSource source)
+        {
+            int sourceNumber = numberedTouchSources[source];
+            return sourceNumber % 2 == 0 ? OsuAction.RightButton : OsuAction.LeftButton;
+        }
```

Comment:
Same as above, `numberedTouchSources` is not required.
```suggestion
        private OsuAction getActionForTouchSource(TouchSource source) => (int)source % 2 == 0 ? OsuAction.RightButton : OsuAction.LeftButton;
```